woocommerce/woocommerce-ios
# Products (M3): Add a new Product Category

Resolves #2000 

## Description
This PR implements the last part for completing the management of Product Categories (part of Products Milestone 3). In particular, in this PR it's possible to create a new product category and to select a parent category.

## Changes
- The `addCloseNavigationBarButton:` method helper now accept also a custom title.
- Implemented the `AddProductCategoryViewController`
- Implemented the `ProductParentCategoriesViewController`
- `ProductCategoryListViewController` now contains the new “Add Button” on top of the view.
- Implemented a new method `addAndSelectedNewCategory:` in `ProductCategoryListViewModel`
- `TextFieldTableViewCell` now contains a new method `resignFirstResponder:`

## Testing
1. Open a product with or without product categories
2. Open the Product Categories screen
3. Tap "Add Category" button on top
4. Try to create a new category without parent category -> When it's saved, you will see it in category list selected. Save the product and make sure that it will be updated correctly.
5. Try to create a new category with parent category -> The new parent category should be selected in "New category" screen. Save the product and make sure that it will be updated correctly.
6. Try to create a new category where the title it's just a space ` `. -> Make sure to see an error notice which appears on the screen, and where you are able to retry.


## Add Product Category in action (GIF)
![add product category](https://user-images.githubusercontent.com/495617/85430090-f4c13c00-b57f-11ea-93bf-6b2842b287f6.gif)


## Screenshots (dark mode enabled)
| Categories             |  Add Category | Select Parent Category |
:-------------------------:|:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 11 Pro - 2020-06-23 at 18 31 01](https://user-images.githubusercontent.com/495617/85430042-e6732000-b57f-11ea-9efa-ac63631384dc.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-06-23 at 18 31 12](https://user-images.githubusercontent.com/495617/85430053-e96e1080-b57f-11ea-92c9-c067894884f4.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-06-23 at 18 31 17](https://user-images.githubusercontent.com/495617/85430061-eb37d400-b57f-11ea-87fb-65b26043b391.png)


Update release notes:
